File: server/routes/booking_routes.py
from flask import Blueprint
from controllers.booking_controller import create_booking, get_user_bookings, get_booking_by_id, cancel_booking
from middleware.auth_middleware import token_required

booking_bp = Blueprint('booking_bp', __name__)

booking_bp.route('', methods=['POST'])(token_required(create_booking))
booking_bp.route('', methods=['GET'])(token_required(get_user_bookings))
booking_bp.route('/my', methods=['GET'])(token_required(get_user_bookings))
booking_bp.route('/<int:booking_id>', methods=['GET'])(token_required(get_booking_by_id))
booking_bp.route('/<int:booking_id>/cancel', methods=['POST'])(token_required(cancel_booking))

File: server/routes/bookingRoutes.js
const express = require('express');
const router = express.Router();
const bookingController = require('../controllers/bookingController');
const authMiddleware = require('../middleware/authMiddleware');

router.post('/', (req, res, next) => {
  if (req.headers.authorization) {
    return authMiddleware(req, res, () => bookingController.createBooking(req, res, next));
  }
  return bookingController.createBooking(req, res, next);
});
router.get('/', (req, res, next) => {
  if (req.headers.authorization) {
    return authMiddleware(req, res, () => bookingController.getUserBookings(req, res, next));
  }
  return bookingController.getUserBookings(req, res, next);
});
router.get('/my', (req, res, next) => {
  if (req.headers.authorization) {
    return authMiddleware(req, res, () => bookingController.getUserBookings(req, res, next));
  }
  return bookingController.getUserBookings(req, res, next);
});
router.get('/vendor', (req, res, next) => {
  if (req.headers.authorization) {
    return authMiddleware(req, res, () => bookingController.getVendorBookings(req, res, next));
  }
  return bookingController.getVendorBookings(req, res, next);
});
router.get('/:id', bookingController.getBookingById);
router.put('/:id/status', (req, res, next) => {
  if (req.headers.authorization) {
    return authMiddleware(req, res, () => bookingController.updateBookingStatus(req, res, next));
  }
  return bookingController.updateBookingStatus(req, res, next);
});
router.post('/:id/cancel', bookingController.cancelBooking);

module.exports = router;
